scwx-qt/tools/update_radar_sites.py

```diff
@@ -0,0 +1,281 @@
+        stationDict["type"]    = NOAA_TYPE_DICT[station["platforms"][0]["platform"]]
+        #stationDict["tz"]      = station["location"]["geoInfo"]["utcOffsets"][0]["utcOffset"] # This is UTC offset, not timezone
+        """
+
+        output[stationId] = stationDict
+
+    return output
+
+# Get the list of updated stations (not in place), using the noaaStationsDict
+# from make_noaa_stations_dict
+def update_stations(noaaStationsDict, previousStations):
+    newStations = []
+    for station in previousStations:
+        newStation = station.copy()
+
+        if not station["id"] in noaaStationsDict:
+            # may be good idea to add fallback to a ICAO search for non active
+            # stations.
+            print(f"WARNING: Station '{station['id']}' not found in noaa data")
+
+            if "elevation" not in station:
+                newStation["elevation"] = None
+        else:
+            newStation.update(noaaStationsDict[station["id"]])
+
+        newStations.append(newStation)
+
+    return newStations
+
+# Customized dump routine. Formats it as one station per row, aligning items.
+def custom_dump(stations, file):
+    file.write("[\n")
+    lengths = {}
+    lastKey = None
+    keys = None
+
+    # Find length for each value, and ensure all stations have the same keys.
+    for station in stations:
+        for key, value in station.items():
+            length = len(json.dumps(value))
+            if key in lengths:
+                lengths[key] = max(length, lengths[key])
+            else:
+                lengths[key] = length
+            lastKey = key
+
+        newKeys = list(station.keys())
+        if keys is None:
+            keys = newKeys
+        elif keys != newKeys:
+            print("DUMP ERROR: Stations did not have the same keys.")
+            exit(3)
+
+    # Write out each station with the correct format.
+    lastType = None
+    for station in stations:
+        # put an empty line between NEXRAD and TDWR.
+        if lastType is not None and lastType != station["type"]:
+            file.write("\n")
+
+        file.write("\t{ ")
+
+        for key, value in station.items():
+            value = json.dumps(value)
+            file.write(f'"{key}": {value}')
+
+            if key != lastKey:
+                file.write(", ")
+                file.write(" " * (lengths[key] - len(value)))
+
+        if station == stations[-1]:
+            file.write(" }\n")
+        else:
+            file.write(" },\n")
+
+        lastType = station["type"]
+
+    file.write("]\n")
+
+# Write coordinates out to a file. Useful for checking against map program.
+def make_coords(stations, file):
+    for station in stations:
+        lat = str(abs(station["lat"]))
+        lat += "N" if station["lat"] > 0 else "S"
+
+        lon = str(abs(station["lon"]))
+        lon += "E" if station["lon"] > 0 else "W"
+
+        file.write(f"{lat} {lon}\n")
+
+def main():
+    parser = argparse.ArgumentParser(
```

Comment:
It looks like this could be made a CMake target, similar to other Python scripts.

https://github.com/dpaulat/supercell-wx/blob/a7a30347aed962f9e33654b54c5f8091b842cb33/scwx-qt/scwx-qt.cmake#L476-L489

I wouldn't make this a dependency of scwx-qt or supercell-wx, however, like the existing scripts are.

https://github.com/dpaulat/supercell-wx/blob/a7a30347aed962f9e33654b54c5f8091b842cb33/scwx-qt/scwx-qt.cmake#L491

Not holding up the pull request, just a thought.

Reply:
I am not super familiar with CMake, but I think I got it set up correctly. I only used `add_custom_target` so that it would not be run on builds (because `radar_sites.json` is a dependency), and so `radar_sites.json` would not be marked as GENERATED, which I believe would make the clean target remove it, which would break this script. I am able to build supercell-wx and update the radar sites separately. 
I also added a warning to the script about it possibly breaking the tests in `radar_site.test.cpp` (just a printed warning using ANSI colors). I attached a screenshot of the warning below. If the text or color should be changed or the warning removed, I can do that fairly quickly.
![supercellwx-warning-screenshoot](https://github.com/dpaulat/supercell-wx/assets/38887432/198dceae-6753-4260-9442-57a162d15fbb)
"WARNING: Updating radar sites may break tests in 'test/source/scwx/qt/config/radar_site.test.cpp'"